File: Universal Windows Platform Solution/Phantom of the West/Phantom of the West/Visual Novel/Chapters/ChapterSelector.cs
using Phantom_of_the_West.Visual_Novel.Chapters.Chapter_0;
using Phantom_of_the_West.Visual_Novel.Chapters.Chapter_1;
using Phantom_of_the_West.Visual_Novel.Chapters.Chapter_10;
using Phantom_of_the_West.Visual_Novel.Chapters.Chapter_11;
using Phantom_of_the_West.Visual_Novel.Chapters.Chapter_12;
using Phantom_of_the_West.Visual_Novel.Chapters.Chapter_13;
using Phantom_of_the_West.Visual_Novel.Chapters.Chapter_14;
using Phantom_of_the_West.Visual_Novel.Chapters.Chapter_15;
using Phantom_of_the_West.Visual_Novel.Chapters.Chapter_16;
using Phantom_of_the_West.Visual_Novel.Chapters.Chapter_17;
using Phantom_of_the_West.Visual_Novel.Chapters.Chapter_2;
using Phantom_of_the_West.Visual_Novel.Chapters.Chapter_3;
using Phantom_of_the_West.Visual_Novel.Chapters.Chapter_4;
using Phantom_of_the_West.Visual_Novel.Chapters.Chapter_5;
using Phantom_of_the_West.Visual_Novel.Chapters.Chapter_6;
using Phantom_of_the_West.Visual_Novel.Chapters.Chapter_7;
using Phantom_of_the_West.Visual_Novel.Chapters.Chapter_8;
using Phantom_of_the_West.Visual_Novel.Chapters.Chapter_9;
using System.Collections.Generic;

namespace Phantom_of_the_West.Visual_Novel.Chapters
{
	internal delegate StoryChoiceList StoryChoiceListDelegate();

	internal class ChapterSelector
	{
		private static Dictionary<int, IChapter> idToChapter = new Dictionary<int, IChapter>()
		{
			{0, new Chapter0()},
			{1, new Chapter1()},
			{2, new Chapter2()},
			{3, new Chapter3()},
			{4, new Chapter4()},
			{5, new Chapter5()},
			{6, new Chapter6()},
			{7, new Chapter7()},
			{8, new Chapter8()},
			{9, new Chapter9()},
			{10, new Chapter10()},
			{11, new Chapter11()},
			{12, new Chapter12()},
			{13, new Chapter13()},
			{14, new Chapter14()},
			{15, new Chapter15()},
			{16, new Chapter16()},
			{17, new Chapter17()}
		};

		internal static IStoryChoiceList GoToState(int id)
		{
			const int chapterMaxPages = 1000;
			int chapterNumber = id / chapterMaxPages;
			IStoryChoiceList nextList = null;
			if (idToChapter.ContainsKey(chapterNumber))
			{
				IChapter chapter = idToChapter[chapterNumber];
				nextList = chapter.GoToState(id);
			}
			return nextList;
		}
	}
}

File: Android Project/app/src/main/java/com/christopherraleigh/phantomofthewest/visual_novel/chapters/ChapterSelector.java
package com.christopherraleigh.phantomofthewest.visual_novel.chapters;

import android.util.SparseArray;

import com.christopherraleigh.phantomofthewest.visual_novel.IStoryChoiceList;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_0.Chapter0;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_1.Chapter1;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_10.Chapter10;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_11.Chapter11;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_12.Chapter12;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_13.Chapter13;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_14.Chapter14;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_15.Chapter15;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_16.Chapter16;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_17.Chapter17;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_18.Chapter18;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_2.Chapter2;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_3.Chapter3;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_4.Chapter4;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_5.Chapter5;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_6.Chapter6;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_7.Chapter7;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_8.Chapter8;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_9.Chapter9;

/**
 * Created by christopherraleigh on 2016-04-03.
 */
public class ChapterSelector {
    private final static int CHAPTER_MAX_PAGES = 1000;

    private static SparseArray<Chapter> idToChapter =
            new SparseArray<Chapter>() {{
                put(0, new Chapter0());
                put(1, new Chapter1());
                put(2, new Chapter2());
                put(3, new Chapter3());
                put(4, new Chapter4());
                put(5, new Chapter5());
                put(6, new Chapter6());
                put(7, new Chapter7());
                put(8, new Chapter8());
                put(9, new Chapter9());
                put(10, new Chapter10());
                put(11, new Chapter11());
                put(12, new Chapter12());
                put(13, new Chapter13());
                put(14, new Chapter14());
                put(15, new Chapter15());
                put(16, new Chapter16());
                put(17, new Chapter17());
                put(18, new Chapter18());
            }};

    public static IStoryChoiceList goToState(int id) {
        int chapterNumber = id / CHAPTER_MAX_PAGES;
        IStoryChoiceList nextList = null;
        IChapter chapter = idToChapter.get(chapterNumber);
        if (chapter != null) {
            nextList = chapter.goToState(id);
        }
        return nextList;
    }
}
